src/backends/policy_enforcement/souffle/value_test.cc

```diff
@@ -0,0 +1,172 @@
+//-----------------------------------------------------------------------------
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//----------------------------------------------------------------------------
+
+#include "src/backends/policy_enforcement/souffle/value.h"
+
+#include <limits>
+
+#include "absl/strings/numbers.h"
+#include "absl/strings/string_view.h"
+#include "src/common/testing/gtest.h"
+
+namespace raksha::backends::policy_enforcement::souffle {
+
+using testing::Combine;
+using testing::TestWithParam;
+using testing::ValuesIn;
+
+class NumberTest : public TestWithParam<int64_t> {};
+
+TEST_P(NumberTest, NumberTest) {
+  int64_t num = GetParam();
+  Number number_value = Number(num);
+  std::string datalog_str = number_value.ToDatalogString();
+  int64_t parsed_int = 0;
+  bool conversion_succeeds = absl::SimpleAtoi(datalog_str, &parsed_int);
+  ASSERT_TRUE(conversion_succeeds);
+  ASSERT_EQ(parsed_int, num);
+}
+
+static int64_t kSampleIntegerValues[] = {0, -1, 1,
+                                         std::numeric_limits<long>::max(),
+                                         std::numeric_limits<long>::min()};
+
+INSTANTIATE_TEST_SUITE_P(NumberTest, NumberTest,
+                         ValuesIn(kSampleIntegerValues));
+
+class SymbolTest : public TestWithParam<absl::string_view> {};
+
+TEST_P(SymbolTest, SymbolTest) {
+  absl::string_view symbol = GetParam();
+  Symbol symbol_value = Symbol(symbol);
+  std::string symbol_str = symbol_value.ToDatalogString();
+  ASSERT_EQ(symbol_str, "\"" + std::string(symbol) + "\"");
+}
+
+static absl::string_view kSampleSymbols[] = {"", "x", "foo", "hello_world"};
```

Comment:
is `""` a valid symbol?

Reply:
Yes. `Symbol` in Souffle datalog really just means string, so you can have the empty symbol and it will work as you might expect.